doc/hooks-config.rst

```diff
@@ -102,6 +102,181 @@ version 3.0 of Gtk and version 4 of GtkSource:
 
 .. _matplotlib hook options:
 
+
+GStreamer (gi.repository.Gst) hook
+----------------------------------
+
+The collection of GStreamer is subject to both the general ``gi`` hook
+configuration (for example, collection of translations file as controlled
+by the ``languages`` option) and by special hook configuration named
+``gstreamer`` [#gstreamer_id]_ that controls collection of GStreamer plugins.
+
+The GStreamer framework comes with a multitude of plugins that are
+typically installed as separate packages (``gstreamer-plugins-base``,
+``gstreamer-plugins-good``, ``gstreamer-plugins-bad``, and
+``gstreamer-plugins-ugly``; the naming varies between packaging systems).
+By default, PyInstaller collects *all* available plugins as well as their
+binary dependencies; therefore, having all GStreamer plugins installed
+in the build environment will likely result in collection of many
+unnecessary plugins and increased frozen application size due to
+potential complex dependency chains of individual plugins and the
+underlying shared libraries.
+
+**Hook identifier:** ``gstreamer`` [#gstreamer_id]_
+
+**Options**
+
+ * ``include_plugins`` [*list of strings*]: list of plugin names to
+   include in the frozen application. Specifying the include list
+   implicitly excludes all plugins that do not appear in the list.
+
+ * ``exclude_plugins`` [*list of strings*]: list of plugin names to
+   exclude from the frozen application. If include list is also available,
+   the exclude list is applied after it; if not, the exclude list is
+   applied to all available plugins.
+
+Both include and exclude list expect base plugin names (e.g., ``audioparsers``,
+``matroska`` , ``x264``, ``flac``). Internally, each name is converted into
+a pattern (e.g., ``'**/*flac.*'``), and matched using ``fnmatch`` against
+actual plugin file names. Therefore, it is also possible to include the
+wildcard (``*``) in the plugin name [#gstreamer_plugin_prefix]_.
+
+**Basic example: excluding an unwanted plugin**
+
+Exclude the ``opencv`` GStreamer plugin to prevent pulling OpenCV shared
+libraries into the frozen application.
+
+.. code-block:: python
+
+    a = Analysis(
+        ["my-gstreamer-app.py"],
+        ...,
+        hooksconfig={
+            "gstreamer": {
+                "exclude_plugins": [
+                    "opencv",
+                ],
+            },
+        },
+        ...,
+    )
+
+**Advanced example: including only specific plugins**
+
+When optimizing the frozen application size, it is often more effecient
+to explicitly include only the subset of the plugins that are actually
+required for the application to function.
+
+Consider the following simple player application:
+
+.. code-block:: python
+
+    # audio_player.py
+    import sys
+    import os
+
+    import gi
+    gi.require_version('Gst', '1.0')
+    from gi.repository import GLib, Gst
+
+
+    if __name__ == '__main__':
```

Comment:
Subjective point but IMO, if you've not got reusable functions defined in a script then you might as well ditch the `if __name__ == "__main__":` guard.

Reply:
Yeah, I've been on the fence regarding that one... Changed.